forte/processors/query_creator.py

```diff
@@ -29,32 +31,43 @@
 
 
 class QueryCreator(MultiPackProcessor):
-    r"""This processor is used to search for relevant documents for a query
-    """
+    r"""This processor searches relevant documents for a query"""
```

Comment:
Maybe we can also have an abstract QueryProcessor that exposes a ``build_query`` function, and a specific query processor can build the actual query.

Reply:
I have created `QueryProcessor` base class in ebab198